pydantic/main.py

```diff
@@ -799,11 +807,16 @@ def validate_model(  # noqa: C901 (ignore complexity)
     check_extra = config.extra is not Extra.ignore
     cls_ = cls or model
 
+    if loc is None:
+        loc = tuple()
+    elif not isinstance(loc, tuple):
+        loc = (loc, )
+
     for validator in model.__pre_root_validators__:
         try:
             input_data = validator(cls_, input_data)
         except (ValueError, TypeError, AssertionError) as exc:
-            return {}, set(), ValidationError([ErrorWrapper(exc, loc=ROOT_KEY)], cls_)
+            return {}, set(), ValidationError([ErrorWrapper(exc, loc=loc + (ROOT_KEY,))], cls_)
```

Comment:
it might be more performant to add a method

`def _build_loc(current_loc, new_key):
    if current_loc:
        return current_loc + (new_key,)
    else:
        return new_key  # or (new_key,)
```

Reply:
Yeah, there is still some optimization to be done in the hot path here.

What I wanted to know about root validator locs is: Should the root loc of nested models actually contain the ROOT_KEY or should they be claled simply by the name of the field that nested model is in? In the case highlighted here it would mean simply passing in the loc when not none or empty and only passing the ROOT:KEY otherwise.

Reply:
It's important that loc as shown in exceptions should not change from what it is now.

I guess the point in this change is to allow that same loc value that would appear in the exception to be available to the validator.